src/python/pants/option/options.py

```diff
@@ -345,20 +345,6 @@ def register(self, scope: str, *args, **kwargs) -> None:
         if deprecated_scope:
             self.get_registrar(deprecated_scope).register(*args, **kwargs)
 
-    def registration_function_for_subsystem(self, subsystem_cls):
-        """Returns a function for registering options on the given scope."""
-
-        # TODO(benjy): Make this an instance of a class that implements __call__, so we can
-        # docstring it, and so it's less weird than attaching properties to a function.
-        def register(*args, **kwargs):
-            self.register(subsystem_cls.options_scope, *args, **kwargs)
-
-        # Clients can access the bootstrap option values as register.bootstrap.
-        register.bootstrap = self.bootstrap_option_values()
```

Comment:
Since the only invokers of `register` are internal, below, and we know they are not reading these properties, we don't need to set them, and can in fact get rid of this entire function.